### PR TITLE
Initial documentation of query workflow

### PR DESCRIPTION
**Pull request by netopyr**

### CODE REVIEW

**Comment by hendrikebbers** [comment]
I just tried to open the PNG by using draw.io. That was not possible at all. Have you used a different tool to create the diagram?

**Comment by netopyr** [comment]
> I just tried to open the PNG by using draw.io. That was not possible at all. Have you used a different tool to create the diagram?

The source is the SVG file. I don't think it is possible to edit PNG with draw.io. I exported the PNG file and used it in the markdown, because in the past there were issues using SVG-files in markdown. (Maybe these have been resolved and I can stop exporting PNGs. Have not checked it lately.)

**Review by mhess-swl — APPROVED** [review]
Looks good!

File: docs/design/app/workflows.md [hunk 1]
```diff
@@ -47,4 +48,33 @@ An `Event` at a time is sent to the `prehandle` with a reference to the latest i
...
+The query-workflow is single-threaded, but multiple calls can run in parallel.
+
+The query workflow consists of the following steps:
+
+1. **Parse and check header.** The query arrives as a byte-array. The required parts are parsed and the type of the query (header) is extracted and validated.
+2. **Check query throttles.** Throttling must be observed and checked as early as possible.
+3. If a payment is required:
+   1. **Validate the CryptoTransfer.** The contained `CryptoTransfer` is validated.
+   2. **Check account balances.** The accounts of all payers are checked.
+4. **Check permissions.** It is checked, if the requester is actually allowed to do the query.
```
Comment (Neeharika-Sompalli) [inline docs/design/app/workflows.md:74]: May be it is already discussed. Is there a reason to not do the `Permissions` and `Validity` after throttles in the picture ?

**Comment by hendrikebbers** [comment]
> I don't think it is possible to edit PNG with draw.io.

"Ensure Include a copy of my diagram is enabled to be able to open your diagram and edit it when you import the PNG file to diagrams.net." I learned that some time ago. Very cool
uuzQfsCyhbE09I-zbbtC7HN05uP4mMup8Q

File: docs/design/app/workflows.md (hunk 1 above)
Comment (netopyr) [reply docs/design/app/workflows.md:74]: This is more or less the order in which these things are done today. We did not discuss any changes (yet).

My guess is, that we want to ensure the payer can pay for the query before doing any further work.

**Review by mhess-swl — APPROVED** [review]

File: docs/design/app/workflows.md (hunk 1 above)
Comment (netopyr) [reply docs/design/app/workflows.md]: After our discussion last week, I updated the documentation accordingly.

**Review by mhess-swl — APPROVED** [review]

**Review by hendrikebbers — APPROVED** [review]

**Pull request merged**